### PR TITLE
[api] Use folk java process to avoid jvm consume GPU memory

### PR DESCRIPTION
**Pull request by frankfliu**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: api/src/main/java/ai/djl/util/cuda/CudaUtils.java [hunk 1]
```diff
@@ -155,8 +188,42 @@ public static MemoryUsage getGpuMemory(Device device) {
...
+    @SuppressWarnings("PMD.SystemPrintln")
+    public static void main(String[] args) {
+        int gpuCount = getGpuCount();
+        if (args.length == 0) {
+            if (gpuCount <= 0) {
+                System.out.println("0,-1");
+                return;
+            }
+            int cudaVersion = getCudaVersion();
+            System.out.println(gpuCount + "," + cudaVersion);
```
Comment (xyang16) [inline api/src/main/java/ai/djl/util/cuda/CudaUtils.java:205]: Why use println?

Comment (frankfliu) [reply api/src/main/java/ai/djl/util/cuda/CudaUtils.java:205]: We need to read the console output from java side

File: api/src/test/java/ai/djl/util/cuda/CudaUtilsTest.java [hunk 2]
```diff
@@ -30,23 +28,34 @@ public class CudaUtilsTest {
...
 
-        List<String> supportedSm = Arrays.asList("37", "52", "60", "61", "70", "75");
-        Assert.assertTrue(supportedSm.contains(smVersion), "Unsupported cuda sm: " + smVersion);
+    @Test
+    public void testCudaUtilsWithFolk() {
+        System.setProperty("ai.djl.util.cuda.folk", "true");
+        try {
+            testCudaUtils();
+        } finally {
+            System.clearProperty("ai.djl.util.folk_cuda_utils");
```
Comment (zachgk) [inline api/src/test/java/ai/djl/util/cuda/CudaUtilsTest.java:58]:
```suggestion
            System.clearProperty("ai.djl.util.folk");
```

File: api/src/main/java/ai/djl/util/cuda/CudaUtils.java [hunk 3]
```diff
@@ -50,6 +54,11 @@ public static boolean hasCuda() {
      * @return the number of GPUs available in the system
      */
     public static int getGpuCount() {
+        if (Boolean.getBoolean("ai.djl.util.cuda.folk")) {
+            String[] ret = execute(-1);
```
Comment (zachgk) [inline api/src/main/java/ai/djl/util/cuda/CudaUtils.java:58]: Can we cache the result of execute?

File: api/src/test/java/ai/djl/util/cuda/CudaUtilsTest.java (hunk 2 above)
Comment (frankfliu) [reply api/src/test/java/ai/djl/util/cuda/CudaUtilsTest.java]: Fixed

File: api/src/main/java/ai/djl/util/cuda/CudaUtils.java (hunk 3 above)
Comment (frankfliu) [reply api/src/main/java/ai/djl/util/cuda/CudaUtils.java]: Fixed

**Review by xyang16 — APPROVED** [review]

**Pull request merged**